gwt-jackson/src/main/java/com/github/nmorel/gwtjackson/client/utils/DateFormat.java

```diff
@@ -173,24 +173,30 @@ public static Date parse( String date ) {
     /**
      * Parse a date using the pattern given in parameter or {@link #DATE_FORMAT_STR_ISO8601} and the browser timezone.
      *
+     * @param adjustDatesToContextTimeZone wheter the date needs to adjusted to current timezone.
      * @param pattern pattern to use. If null, {@link #DATE_FORMAT_STR_ISO8601} will be used.
+     * @param patternHasTz whether the pattern includes timezone information. when null the pattern will be parsed to search it.
      * @param date date to parse
      *
      * @return the parsed date
      */
-    public static Date parse( String pattern, String date ) {
+    public static Date parse( boolean adjustDatesToContextTimeZone, String pattern, Boolean patternHasTz, String date ) {
         if ( null == pattern ) {
             return parse( DateFormat.DATE_FORMAT_STR_ISO8601, date );
         } else {
-            DateParser parser = CACHE_PARSERS.get( pattern );
+            String patternCacheKey = pattern + adjustDatesToContextTimeZone;
```

Comment:
It's necessary to add the adjustDatesToContextTimeZone to the cache key for the case when you use once adjustDatesToContextTimeZone enabled and the in the same app flag disabled.